pkg/controller/kogitoapp/build/trigger.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build
+
+import (
+	"fmt"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/openshift"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/util"
+	buildv1 "github.com/openshift/api/build/v1"
+)
+
+const (
+	buildConfigLabelSelector = "buildconfig"
+)
+
+// Trigger defines how to interact with build triggers in a OpenShift cluster
+type Trigger interface {
+	// SelectOneBuildConfigWithLabel restricts the trigger only on the first build with the given labels
+	SelectOneBuildConfigWithLabel(labelSelector map[string]string, buildConfigs ...*buildv1.BuildConfig) Trigger
+	// OnBuildConfig defines the build configuration to trigger the build.
+	// Another approach is using `SelectOneBuildConfigWithLabel` to select the build to trigger
+	OnBuildConfig(buildConfig *buildv1.BuildConfig) Trigger
+	// StartNewBuildIfNotRunning starts a new build for the filtered build config
+	StartNewBuildIfNotRunning() (result TriggerResult, err error)
+	// HasBuildConfiguration returns true if a build configuration was selected for this instance
+	HasBuildConfiguration() bool
+}
+
+// TriggerResult structure to hold build trigger information
+type TriggerResult struct {
+	Started   bool
+	BuildName string
+}
+
+type trigger struct {
+	triggeredBy   string
+	labelSelector map[string]string
```

Comment:
This parameter isn't used.

Reply:
yeah, I was using it on an other alternative, forget to remove it.